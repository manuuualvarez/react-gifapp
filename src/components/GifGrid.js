import React from 'react'
// Components
import {  GifGridItem } from './GifGridItem'
// Hooks
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ( {category }) => {

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> { category } </h3>
        
            { loading && <p className="animate__animated animate__flash">Loading...</p>  }

            <div className="card-grid">
                {  images.map( img =>  

                    <GifGridItem  
                        key={ img.id }
                        { ...img }
                    />       
                    )
                }
            </div>
        </>
    )
}
