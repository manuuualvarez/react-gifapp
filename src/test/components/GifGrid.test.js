import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'; 
import {  PropTypes } from 'prop-types';

// Components
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Unit Test on <GifGrid />', () => {

    const category = 'Miami'
    
    test('Show component successfully without request', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category= {category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show items when the server response', () => {
        // Mock data
        const gifs = [
            {id: '01', title: 'Title 1', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '02', title: 'Title 2', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '03', title: 'Title 3', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '04', title: 'Title 4', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '05', title: 'Title 5', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '06', title: 'Title 6', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '07', title: 'Title 7', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '08', title: 'Title 8', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '09', title: 'Title 9', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
            {id: '10', title: 'Title 10', url: 'https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif'},
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category= {category} />);
        // Chweck Snapshot
        expect(wrapper).toMatchSnapshot();
        // Not must show paragraph
        expect(wrapper.find('p').exists()).toBe(false);
        // Check items
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});

GifGrid.propTypes = { 
    category: PropTypes.string.isRequired,
}