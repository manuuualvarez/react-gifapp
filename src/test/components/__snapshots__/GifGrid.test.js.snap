// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test on <GifGrid /> Should show items when the server response 1`] = `
<Fragment>
  <h3>
     
    Miami
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="01"
      key="01"
      title="Title 1"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="02"
      key="02"
      title="Title 2"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="03"
      key="03"
      title="Title 3"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="04"
      key="04"
      title="Title 4"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="05"
      key="05"
      title="Title 5"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="06"
      key="06"
      title="Title 6"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="07"
      key="07"
      title="Title 7"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="08"
      key="08"
      title="Title 8"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="09"
      key="09"
      title="Title 9"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
    <GifGridItem
      id="10"
      key="10"
      title="Title 10"
      url="https://media1.giphy.com/media/l0HlW9jd0y8Vy9qkU/giphy.gif"
    />
  </div>
</Fragment>
`;

exports[`Unit Test on <GifGrid /> Show component successfully without request 1`] = `
<Fragment>
  <h3>
     
    Miami
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
